pipdeptree.py

```diff
@@ -376,7 +379,13 @@ def filter(self, include, exclude):
                                 continue
                 else:
                     break
-
+        assert not exclude & set([n.key for n in m])
+        if exclude_re:
+            _m = {}
+            for n, l in m.items():
```

Comment:
IMO it's better to have this logic inside the above `for..while` loops itself instead of looping through data multiple times. To keep the code concise, you may introduce functions `is_excluded` and `is_included` within this method or as static methods of the class. For eg. `is_excluded` can take 3 args - 

1. exclude (set | None)
2. exclude_re (regex pattern | None) 
3. node key

and can return True or False. I think `exclude` (set) should get higher priority than `exclude_re` but I'm open to discussion if you think the other way around is better.

Reply:
If you favor tidy code over performance then a helper function is fine. I will tell you that with 100's of packages the execution is already quite slow